### PR TITLE
Adding Typesafe Config enhancers.

### PR DESCRIPTION
**Pull request by kshakir**
Also initializing build.sbt, README.md, Travis, and Coveralls

### CODE REVIEW

File: src/main/scala/lenthall/util/ScalaConfig.scala [hunk 1]
```diff
@@ -0,0 +1,49 @@
+package lenthall.util
+
+import com.typesafe.config.Config
+
+object ScalaConfig {
+
+  implicit class PimpedScalaConfig(val config: Config) extends AnyVal {
```
Comment (mcovarr) [inline src/main/scala/lenthall/util/ScalaConfig.scala]: `EnhancedScalaConfig`?

File: build.sbt [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  SLF4J: com.imageworks.common.concurrent.SingleThreadInfiniteLoopRunner
+
+As a workaround, load SLF4J's root logger before starting the unit tests
+
+Source: https://github.com/typesafehub/scalalogging/issues/23#issuecomment-17359537
+References:
+  http://stackoverflow.com/a/12095245
+  http://jira.qos.ch/browse/SLF4J-167
+  http://jira.qos.ch/browse/SLF4J-97
+*/
```
Comment (mcovarr) [inline build.sbt]: cool!

File: src/main/scala/lenthall/util/ScalaConfig.scala [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+    def getLongOption(key: String): Option[Long] = {
+      if (config.hasPath(key)) Option(config.getLong(key)) else None
+    }
+
+    def getDoubleOption(key: String): Option[Double] = {
+      if (config.hasPath(key)) Option(config.getDouble(key)) else None
+    }
+
+    def getStringOr(key: String, default: => String = ""): String = {
```
Comment (mcovarr) [inline src/main/scala/lenthall/util/ScalaConfig.scala]: Could these "or" methods all chain to their "option" peers?

```scala
getStringOption(key).getOrElse(default)
```

File: src/main/scala/lenthall/util/ValidatedConfig.scala [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+  implicit class EnhancedValidatedConfig(val config: Config) extends AnyVal {
+    def keys: Set[String] = {
+      config.entrySet().toSet map { entry: java.util.Map.Entry[String, ConfigValue] => entry.getKey }
+    }
+
+    /**
+     * For keys that are in the configuration but not in the reference keySet, log a warning.
+     */
+    def warnNotRecognized(keySet: Set[String], context: String) = {
+      keys.diff(keySet) match {
```
Comment (mcovarr) [inline src/main/scala/lenthall/util/ValidatedConfig.scala]: isn't this just
```scala
if (keys.diff(keySet).nonEmpty) {...
```

**Comment by mcovarr** [comment]
Looks good!  :+1:

File: build.sbt [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+name := "lenthall"
+
+version := "0.13-SNAPSHOT"
+
+organization := "org.broadinstitute"
+
+scalaVersion := "2.11.7"
+
+libraryDependencies ++= Seq(
+  "com.typesafe" % "config" % "1.2.1",
```
Comment (geoffjentry) [inline build.sbt]: Would another project using lenthall (e.g. Tyburn) need to include these in their own SBT now or is this considered part of the common code?

File: src/main/scala/lenthall/util/ScalaConfig.scala (hunk 3 above)
Comment (kshakir) [inline src/main/scala/lenthall/util/ScalaConfig.scala]: Interesting… I'm torn at the moment… that would be slightly fewer characters and copies of the `if` statements, but would add slightly more object creation and method invocation.

Am I missing anything, and how strong does this offend?

File: build.sbt (hunk 5 above)
Comment (kshakir) [inline build.sbt]: Considered part of the common code. Anything not `Provided` will become a transitive dependency, and will not need to be included.

File: src/main/scala/lenthall/util/ScalaConfig.scala (hunk 3 above)
Comment (scottfrazer) [inline src/main/scala/lenthall/util/ScalaConfig.scala]: FWIW, I think either way is fine

Comment (mcovarr) [inline src/main/scala/lenthall/util/ScalaConfig.scala]: I don't have strong opinions on this, it just seemed that the "option" methods were doing a chunk of the work the "or"s needed, but if you feel the costs outweigh the benefits that's ok. :smile:

**Comment by mcovarr** [comment]
:+1:

File: src/main/scala/lenthall/util/ScalaConfig.scala (hunk 3 above)
Comment (geoffjentry) [inline src/main/scala/lenthall/util/ScalaConfig.scala]: I'd prefer more sugar at the cost of potentially losing performance, regardless of what's being debated

**Comment by scottfrazer** [comment]
:+1:

**Comment by kshakir** [comment]
Comments all addressed. Unless any objections in the next couple mins, squashing and merging.

**Pull request merged**